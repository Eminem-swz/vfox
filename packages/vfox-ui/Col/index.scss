@use 'sass:math';
@import '../style/var.scss';

.fx-col {
  position: relative;
  display: block;
  flex-grow: 0;
  flex-shrink: 0;
  max-width: 100%;
  min-height: 1px;
  box-sizing: border-box;

  &-0 {
    display: none;
  }

  @for $i from 1 through 24 {
    &-#{$i} {
      flex-basis: percentage(math.div($i, 24));
      max-width: percentage(math.div($i, 24));
    }

    &-offset-#{$i} {
      margin-left: percentage(math.div($i, 24));
    }

    &-push-#{$i} {
      left: percentage(math.div($i, 24));
    }

    &-pull-#{$i} {
      right: percentage(math.div($i, 24));
    }
  }
}
