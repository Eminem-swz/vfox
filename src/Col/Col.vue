<template>
  <div :class="classNames" :style="styles">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { defineComponent, inject, computed } from 'vue'
import { isNumeric, rangeInteger } from '@/helpers/util'
import type { StyleObject } from '../helpers/types'

function rangeCol(number: number | string) {
  return rangeInteger(number, 0, 24)
}

export default defineComponent({
  name: 'fx-col',
  props: {
    // 栅格占据的列数
    span: {
      type: [Number, String],
      validator: (val: number | string) => isNumeric(val),
      default: 24
    },
    // 栅格左侧的间隔格数
    offset: {
      type: Number,
      default: 0
    },
    // 栅格向右移动格数
    push: {
      type: Number,
      default: 0
    },
    // 栅格向左移动格数
    pull: {
      type: Number,
      default: 0
    }
  },
  setup(props) {
    const rowGutter = inject('fxRowGutter', [0, 0])

    const styles = computed(() => {
      const styles: StyleObject = {}

      const [gH, gV] = rowGutter

      if (gH > 0 || gV > 0) {
        styles.padding = `${gV / 2}px ${gH / 2}px`
      }

      return styles
    })

    const classNames = computed(() => {
      const arr = [`fx-col`, `fx-col-${rangeCol(props.span)}`]

      if (props.offset > 0) {
        arr.push(`fx-col-offset-${Math.min(24, rangeCol(props.offset))}`)
      }

      if (props.push > 0) {
        arr.push(`fx-col-push-${rangeCol(props.push)}`)
      }

      if (props.pull > 0) {
        arr.push(`fx-col-pull-${rangeCol(props.pull)}`)
      }

      return arr
    })

    return {
      styles,
      classNames
    }
  }
})
</script>
