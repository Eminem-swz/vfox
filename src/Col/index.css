.fx-col {
  position: relative;
  display: block;
  flex-grow: 0;
  flex-shrink: 0;
  max-width: 100%;
  min-height: 1px;
  box-sizing: border-box;
}

.fx-col-0 {
  display: none;
}

.fx-col-1 {
  flex-basis: 4.16667%;
  max-width: 4.16667%;
}

.fx-col-offset-1 {
  margin-left: 4.16667%;
}

.fx-col-push-1 {
  left: 4.16667%;
}

.fx-col-pull-1 {
  right: 4.16667%;
}

.fx-col-2 {
  flex-basis: 8.33333%;
  max-width: 8.33333%;
}

.fx-col-offset-2 {
  margin-left: 8.33333%;
}

.fx-col-push-2 {
  left: 8.33333%;
}

.fx-col-pull-2 {
  right: 8.33333%;
}

.fx-col-3 {
  flex-basis: 12.5%;
  max-width: 12.5%;
}

.fx-col-offset-3 {
  margin-left: 12.5%;
}

.fx-col-push-3 {
  left: 12.5%;
}

.fx-col-pull-3 {
  right: 12.5%;
}

.fx-col-4 {
  flex-basis: 16.66667%;
  max-width: 16.66667%;
}

.fx-col-offset-4 {
  margin-left: 16.66667%;
}

.fx-col-push-4 {
  left: 16.66667%;
}

.fx-col-pull-4 {
  right: 16.66667%;
}

.fx-col-5 {
  flex-basis: 20.83333%;
  max-width: 20.83333%;
}

.fx-col-offset-5 {
  margin-left: 20.83333%;
}

.fx-col-push-5 {
  left: 20.83333%;
}

.fx-col-pull-5 {
  right: 20.83333%;
}

.fx-col-6 {
  flex-basis: 25%;
  max-width: 25%;
}

.fx-col-offset-6 {
  margin-left: 25%;
}

.fx-col-push-6 {
  left: 25%;
}

.fx-col-pull-6 {
  right: 25%;
}

.fx-col-7 {
  flex-basis: 29.16667%;
  max-width: 29.16667%;
}

.fx-col-offset-7 {
  margin-left: 29.16667%;
}

.fx-col-push-7 {
  left: 29.16667%;
}

.fx-col-pull-7 {
  right: 29.16667%;
}

.fx-col-8 {
  flex-basis: 33.33333%;
  max-width: 33.33333%;
}

.fx-col-offset-8 {
  margin-left: 33.33333%;
}

.fx-col-push-8 {
  left: 33.33333%;
}

.fx-col-pull-8 {
  right: 33.33333%;
}

.fx-col-9 {
  flex-basis: 37.5%;
  max-width: 37.5%;
}

.fx-col-offset-9 {
  margin-left: 37.5%;
}

.fx-col-push-9 {
  left: 37.5%;
}

.fx-col-pull-9 {
  right: 37.5%;
}

.fx-col-10 {
  flex-basis: 41.66667%;
  max-width: 41.66667%;
}

.fx-col-offset-10 {
  margin-left: 41.66667%;
}

.fx-col-push-10 {
  left: 41.66667%;
}

.fx-col-pull-10 {
  right: 41.66667%;
}

.fx-col-11 {
  flex-basis: 45.83333%;
  max-width: 45.83333%;
}

.fx-col-offset-11 {
  margin-left: 45.83333%;
}

.fx-col-push-11 {
  left: 45.83333%;
}

.fx-col-pull-11 {
  right: 45.83333%;
}

.fx-col-12 {
  flex-basis: 50%;
  max-width: 50%;
}

.fx-col-offset-12 {
  margin-left: 50%;
}

.fx-col-push-12 {
  left: 50%;
}

.fx-col-pull-12 {
  right: 50%;
}

.fx-col-13 {
  flex-basis: 54.16667%;
  max-width: 54.16667%;
}

.fx-col-offset-13 {
  margin-left: 54.16667%;
}

.fx-col-push-13 {
  left: 54.16667%;
}

.fx-col-pull-13 {
  right: 54.16667%;
}

.fx-col-14 {
  flex-basis: 58.33333%;
  max-width: 58.33333%;
}

.fx-col-offset-14 {
  margin-left: 58.33333%;
}

.fx-col-push-14 {
  left: 58.33333%;
}

.fx-col-pull-14 {
  right: 58.33333%;
}

.fx-col-15 {
  flex-basis: 62.5%;
  max-width: 62.5%;
}

.fx-col-offset-15 {
  margin-left: 62.5%;
}

.fx-col-push-15 {
  left: 62.5%;
}

.fx-col-pull-15 {
  right: 62.5%;
}

.fx-col-16 {
  flex-basis: 66.66667%;
  max-width: 66.66667%;
}

.fx-col-offset-16 {
  margin-left: 66.66667%;
}

.fx-col-push-16 {
  left: 66.66667%;
}

.fx-col-pull-16 {
  right: 66.66667%;
}

.fx-col-17 {
  flex-basis: 70.83333%;
  max-width: 70.83333%;
}

.fx-col-offset-17 {
  margin-left: 70.83333%;
}

.fx-col-push-17 {
  left: 70.83333%;
}

.fx-col-pull-17 {
  right: 70.83333%;
}

.fx-col-18 {
  flex-basis: 75%;
  max-width: 75%;
}

.fx-col-offset-18 {
  margin-left: 75%;
}

.fx-col-push-18 {
  left: 75%;
}

.fx-col-pull-18 {
  right: 75%;
}

.fx-col-19 {
  flex-basis: 79.16667%;
  max-width: 79.16667%;
}

.fx-col-offset-19 {
  margin-left: 79.16667%;
}

.fx-col-push-19 {
  left: 79.16667%;
}

.fx-col-pull-19 {
  right: 79.16667%;
}

.fx-col-20 {
  flex-basis: 83.33333%;
  max-width: 83.33333%;
}

.fx-col-offset-20 {
  margin-left: 83.33333%;
}

.fx-col-push-20 {
  left: 83.33333%;
}

.fx-col-pull-20 {
  right: 83.33333%;
}

.fx-col-21 {
  flex-basis: 87.5%;
  max-width: 87.5%;
}

.fx-col-offset-21 {
  margin-left: 87.5%;
}

.fx-col-push-21 {
  left: 87.5%;
}

.fx-col-pull-21 {
  right: 87.5%;
}

.fx-col-22 {
  flex-basis: 91.66667%;
  max-width: 91.66667%;
}

.fx-col-offset-22 {
  margin-left: 91.66667%;
}

.fx-col-push-22 {
  left: 91.66667%;
}

.fx-col-pull-22 {
  right: 91.66667%;
}

.fx-col-23 {
  flex-basis: 95.83333%;
  max-width: 95.83333%;
}

.fx-col-offset-23 {
  margin-left: 95.83333%;
}

.fx-col-push-23 {
  left: 95.83333%;
}

.fx-col-pull-23 {
  right: 95.83333%;
}

.fx-col-24 {
  flex-basis: 100%;
  max-width: 100%;
}

.fx-col-offset-24 {
  margin-left: 100%;
}

.fx-col-push-24 {
  left: 100%;
}

.fx-col-pull-24 {
  right: 100%;
}
