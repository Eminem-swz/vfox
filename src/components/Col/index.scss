@import '../style/var.scss';

.#{$prefix}-col {
  position: relative;
  display: block;
  flex-grow: 0;
  flex-shrink: 0;
  max-width: 100%;
  min-height: 1px;
  box-sizing: border-box;

  &-0 {
    display: none;
  }

  @for $i from 1 through 24 {
    &-#{$i} {
      flex-basis: percentage($i / 24);
      max-width: percentage($i / 24);
    }

    &-offset-#{$i} {
      margin-left: percentage($i / 24);
    }

    &-push-#{$i} {
      left: percentage($i / 24);
    }

    &-pull-#{$i} {
      right: percentage($i / 24);
    }
  }
}
