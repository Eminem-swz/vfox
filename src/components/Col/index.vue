<template>
  <div :class="classNames" :style="styles" @click="onClick">
    <slot></slot>
  </div>
</template>

<script>
import { SDKKey } from '../../config'
import { rangeInteger } from '../../helpers/util'

function rangeCol(number) {
  return rangeInteger(number, 0, 24)
}

export default {
  name: SDKKey + '-col',
  props: {
    // 栅格占据的列数
    span: {
      type: Number,
      default: 24
    },
    // 栅格左侧的间隔格数
    offset: {
      type: Number,
      default: 0
    },
    // 栅格向右移动格数
    push: {
      type: Number,
      default: 0
    },
    // 栅格向左移动格数
    pull: {
      type: Number,
      default: 0
    }
  },
  data() {
    return {
      gutter: [0, 0],
      prefix: SDKKey
    }
  },
  computed: {
    styles() {
      const arr = []

      const [gH, gV] = this.gutter
      if (gH > 0 || gV > 0) {
        arr.push({
          padding: `${gV / 2}px ${gH / 2}px`
        })
      }

      return arr
    },
    classNames() {
      const arr = [`${SDKKey}-col`, `${SDKKey}-col-${rangeCol(this.span)}`]

      if (this.offset > 0) {
        arr.push(`${SDKKey}-col-offset-${Math.min(24, rangeCol(this.offset))}`)
      }

      if (this.push > 0) {
        arr.push(`${SDKKey}-col-push-${rangeCol(this.push)}`)
      }

      if (this.pull > 0) {
        arr.push(`${SDKKey}-col-pull-${rangeCol(this.pull)}`)
      }

      return arr
    }
  },
  watch: {},
  created() {
    if (this.$parent && this.$parent.gutter2) {
      this.updateGutter(this.$parent.gutter2)
    }
  },
  ready() {},
  mounted() {},
  updated() {},
  attached() {},
  beforeDestroy() {},
  methods: {
    updateGutter(gutter) {
      this.gutter = gutter
    },

    onClick(e) {
      this.$emit(e.type, e)
    }
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-col {
  position: relative;
  display: block;
  flex-grow: 0;
  flex-shrink: 0;
  max-width: 100%;
  min-height: 1px;
  box-sizing: border-box;

  &-0 {
    display: none;
  }

  @for $i from 1 through 24 {
    &-#{$i} {
      flex-basis: percentage($i / 24);
      max-width: percentage($i / 24);
    }

    &-offset-#{$i} {
      margin-left: percentage($i / 24);
    }

    &-push-#{$i} {
      left: percentage($i / 24);
    }

    &-pull-#{$i} {
      right: percentage($i / 24);
    }
  }
}
</style>
