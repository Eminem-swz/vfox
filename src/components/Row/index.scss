@import '../style/var.scss';

.#{$prefix}-row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  box-sizing: border-box;
  justify-content: flex-start;
  align-items: flex-start;

  &.justify {
    &--start {
      justify-content: flex-start;
    }

    &--end {
      justify-content: flex-end;
    }

    &--center {
      justify-content: center;
    }

    &--space-around {
      justify-content: space-around;
    }

    &--space-between {
      justify-content: space-between;
    }
  }

  &.align {
    &--top {
      align-items: flex-start;
    }

    &--middle {
      align-items: center;
    }

    &--bottom {
      align-items: flex-end;
    }
  }
}
