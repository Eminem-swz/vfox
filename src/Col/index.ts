import { withInstall } from '@/helpers/with-install'
import Col from './Col.vue'

export { Col }
export default withInstall(Col)
