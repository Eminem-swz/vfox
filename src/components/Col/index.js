import Col from './Col.vue'

Col.install = function(Vue) {
  Vue.component(Col.name, Col)
}

export default Col
