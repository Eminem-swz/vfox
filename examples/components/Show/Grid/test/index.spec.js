/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Row from '@/Row'
import Col from '@/Col'

describe('Grid', () => {
  test('Row should render default correctly', () => {
    const wrapper = mount(Row, {})

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Col should render default slot correctly', () => {
    const wrapper = mount(Col, {
      slots: {
        default: 'content'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Row & Col should render default correctly', () => {
    const wrapper = mount(Row, {
      slots: {
        default: mount(Col, {
          slots: {
            default: 'content'
          }
        }).html()
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
