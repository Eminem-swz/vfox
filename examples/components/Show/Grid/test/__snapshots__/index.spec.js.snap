// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Col should render default slot correctly 1`] = `<div class="fx-col fx-col-24">content</div>`;

exports[`Grid Row & Col should render default correctly 1`] = `
<div class="fx-row">
  <div class="fx-col fx-col-24">content</div>
</div>
`;

exports[`Grid Row should render default correctly 1`] = `<div class="fx-row"></div>`;
